debuginfo/src/dwarf.rs

```diff
@@ -800,28 +816,12 @@ impl<'d, 'a> DwarfUnit<'d, 'a> {
                                 };
 
                                 lines.insert(index, line_info);
-                                continue;
-                            }
-
-                            // Split the parent record if it exceeds the end of this range. We can
-                            // assume that record.size is set here since we passed the previous
-                            // condition.
-                            let split = if record_end > range_end {
-                                record.size = Some(range_end - record.address);
-
-                                Some(LineInfo {
-                                    address: range_end,
-                                    size: Some(record_end - range_end),
-                                    file: record.file.clone(),
-                                    line: record.line,
-                                })
+                                index += 1;
                             } else {
-                                None
+                                record.file = file.clone();
+                                record.line = line;
```

Comment:
I believe that this still needs to run unconditionally. Otherwise, it looks like the middle record would not have correct file/line info.